torch/onnx/symbolic_opset9.py

```diff
@@ -2666,11 +2666,12 @@ def prim_tolist(g, input, dim_val, elem_ty_val):
     return input
 
 
-@parse_args('v', 'i')
 def one_hot(g, self, num_classes):
     values = g.op("Constant", value_t=torch.LongTensor([0, 1]))
-    depth = g.op("Constant", value_t=torch.LongTensor([num_classes]))
-    return g.op("OneHot", self, depth, values, axis_i=-1)
+    # onnxruntime supports limited type combinations for OneHot.
+    if num_classes.type().scalarType() in ("Byte", "Char", "Int", "Short"):
+        num_classes = g.op("Cast", num_classes, to_i=sym_help.cast_pytorch_to_onnx["Long"])
```

Comment:
curious, will "Cast" be folded if `num_classes` is constant?

Reply:
It should fold, const folding https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/onnx/constant_fold.cpp#L402-L409 takes care of it.